app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -0,0 +1,99 @@
+package com.example.bottomnav;
+
+import org.eclipse.cdt.core.dom.ast.IASTDeclaration;
+import org.eclipse.cdt.core.dom.ast.IASTNode;
+import org.eclipse.cdt.core.dom.ast.IASTTranslationUnit;
+import org.eclipse.cdt.core.dom.ast.gnu.cpp.GPPLanguage;
+import org.eclipse.cdt.core.index.IIndex;
+import org.eclipse.cdt.core.parser.*;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Parse {
+    public HashMap<String, Contents> constRepo = new HashMap<>();
+    public HashMap<String, ArrayList> structRepo = new HashMap<>();
+    private ArrayList subStrucCon = null;
+
+    public Parse(String code) throws Exception {
+        IASTTranslationUnit translationUnit = getIASTTranslationUnit(code.toCharArray());
+        IASTNode node = translationUnit;
+        saveNodes(node);
+    }
+
+    /**Accesses all the nodes of the tree*/
+    private void saveNodes(IASTNode node) throws Exception {
+        IASTNode[] children = node.getChildren();
+        for (IASTNode child : children) {
+
+            if (child instanceof CPPASTSimpleDeclaration) {
+                CPPASTSimpleDeclaration declaration = (CPPASTSimpleDeclaration) child;
+
+                if (declaration.getDeclSpecifier() instanceof CPPASTCompositeTypeSpecifier) {
+                    CPPASTCompositeTypeSpecifier compSpec = (CPPASTCompositeTypeSpecifier) declaration.getDeclSpecifier();
+                    CPPASTName name = (CPPASTName) compSpec.getName();
+                    structHelper(compSpec.getDeclarations(false), name.getRawSignature());
+                } else {
+                    saveAs("const", child);
+                }
+
+            }
+        }
+    }
+
+    private void saveAs(String t, IASTNode node) throws Exception{
```

Comment:
This behaves pretty different in the `const` initializer and `struct` case. I'd suggest making them two different functions.

If you have common code that is used by both, you might refactor that into a helper function that is called by both, for example `getTypeOfDeclaration` or something, to avoid the code duplication that might otherwise happen with two functions.

Note, in some software engineering guides, boolean flags (which `t` here is effectively equivalent to) are sometimes treated with suspicion, because it's a sign that your function might be doing two different things.